assets/app/view/game/buy_trains.rb

```diff
@@ -10,12 +10,15 @@ class BuyTrains < Snabberb::Component
       include EmergencyMoney
       needs :show_other_players, default: nil, store: true
       needs :selected_corporation, default: nil, store: true
+      needs :active_shell, default: nil, store: true
 
       def render_president_contributions
         player = @corporation.owner
 
         children = []
 
+        verb = @must_buy_train ? 'must' : 'may'
```

Comment:
President can contribute funds to buy a train in the shell if the shell is empty